src/Discord/Parts/Channel/Message.php

```diff
@@ -79,6 +80,125 @@ public function reply($text)
     {
         return $this->channel->sendMessage("{$this->author}, {$text}");
     }
+    
+    /**
+     * React to a message.
+     *
+     * @param Emoji $emoticon  The emoticon to react with. (example: '👎', custom: ':michael:251127796439449631')
+     *
+     * @return \React\Promise\Promise
+     */
+    public function react($emoticon)
+    {
+        $deferred = new Deferred();
+
+        $this->http->put(
+            "channels/{$this->channel->id}/messages/{$this->id}/reactions/{$emoticon}/@me"
+        )->then(
+            function ($response) use ($deferred) {
+
+                $deferred->resolve($this);
+            },
+            \React\Partial\bind_right($this->reject, $deferred)
+        );
+
+        return $deferred->promise();
+    }
+    
+    /**
+     * Delete a reaction.
+     *
+     * @param array $settings  (id and emoticon)
+     *
+     * @return \React\Promise\Promise
+     */
+    public function react_delete($type, $emoticon = null, $id = null)
+    {
+        $deferred = new Deferred();
+		
+		$types = ['all', 'me', 'id'];
+		
+		if (in_array($type, $types))
+		{
+			if ($type === 'all')
+			{
+				$url = "channels/{$this->channel->id}/messages/{$this->id}/reactions";
+			}
+			else
+			if ($type === 'me')
+			{
+				$url = "channels/{$this->channel->id}/messages/{$this->id}/reactions/{$emoticon}/@me";
+			}
+			else
+			{
+				$url = "channels/{$this->channel->id}/messages/{$this->id}/reactions/{$emoticon}/{$id}";
+			}
+			
+			$this->http->delete(
+				$url, []
+			)->then(
+				function ($response) use ($deferred) {
+
+					$deferred->resolve($this);
+				},
+				\React\Partial\bind_right($this->reject, $deferred)
+			);
+		}
+		else
+		{
+			$deferred->reject();
+		}
+
+        return $deferred->promise();
+    }
+    
+    /**
+     * Updates the message.
+     *
+     * @param string $text  The text to send in the message.
+     *
+     * @return \React\Promise\Promise
+     */
+    public function edit($text)
```

Comment:
I'd rather keep editing and deleting to the message repositories so I will remove the `edit` and `delete` functions.